.github/workflows/run-tests.yml

```
@@ -100,6 +105,41 @@ jobs:
         run: |
           cat "$GITHUB_OUTPUT"
 
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }}
+
+      - name: Compute merge base
+        run: |
+          MERGE_BASE="$(git merge-base ${{ env.COMMIT }} ${{ github.event.pull_request.base.ref }})"
+          echo "MERGE_BASE=$MERGE_BASE" >> "$GITHUB_ENV"
+
+      - name: Collect Modified Tests
+        id: collect_modified_tests
+        run: |
+          set -e
+          set -o pipefail
```

this is a pretty common bash opener:
```suggestion
          set -euo pipefail
```